staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal_test.go

```diff
@@ -83,6 +244,76 @@ func TestStringArrayOrStringMarshalJSON(t *testing.T) {
 			t.Errorf("Failed to marshal input '%v': expected: %q, got %q", c.input, c.result, string(result))
 		}
 	}
+
+}
+
+func TestJSONSchemaPropsOrBoolMarshalCBOR(t *testing.T) {
```

Comment:
Instead of separate test functions for marshal and unmarshal, would it be better to do all the tests in a single function?  The test cases look almost identical.   Making good use of `t.Run` to organize the tests might be sufficient?

I think this might also make it easier add test cases later and know that we get full coverage since we almost never want to test unmarshal without also test marshal (to ensure round-tripability)

Reply:
I have refactored the tests.